controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ApplicationPackage.java

```diff
@@ -209,6 +219,13 @@ private static ValidationOverrides allValidationOverrides() {
         return ValidationOverrides.fromXml(validationOverridesContents.toString());
     }
 
+    // Hashes all files that require a deployment to be forwarded to configservers
+    private String calculateBundleHash() {
+        Predicate<String> entryMatcher = name -> !name.endsWith(deploymentFile) && !name.endsWith(buildMetaFile);
+        SortedMap<String, Long> entryCRCs = ZipStreamReader.getEntryCRCs(new ByteArrayInputStream(zippedContent), entryMatcher);
+        return String.valueOf(entryCRCs.hashCode());
```

Comment:
Shouldn't we calculate this ourselves? `hashCode()` provides no guarantees across execution and this also depends on the particular implementation of the `SortedMap` that we get.

Reply:
This is true, but as these are strings and numbers, the hash code won't change. They wouldn't dare XD Doesn't hurt to be more careful, of course.